src/gsi/radarZ_module.f90

```diff
@@ -0,0 +1,2889 @@
+
+  !local variables
+   REAL(r_kind) :: gamma_hv, expon
+   REAL(r_kind) :: N0_units
+   REAL(r_double) :: gamma
+
+   gamma_hv = sngl(gamma(dble(beta_a)+dble(beta_b)+dble(alpha) + 1.d0))
+   expon = -(alpha + beta_a + beta_b + 1)
+
+!@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+!
+! Beginning of executable code...
+!
+!@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+
+   N0_units = (1.e-3)**(4.0+alpha)
+
+   refIceHV = mm3todBZ*radar_Const*gamma_hv*(N0*N0_units)*             &
+               (Ci*alp_a**2+Ci*alp_b**2+2*Di*alp_a*alp_b*rho_0)*    &
+               (lamda*1.e-3)**expon
+
+END SUBROUTINE partialRhoIce
+
+SUBROUTINE fractionWater(qr,qi,fo,density_ice,fracqr,fracqi,fm,fw,rhom)
+
+!-----------------------------------------------------------------------
+!
+! PURPOSE:
+!
+! This subroutine calculates the fractions of water, dry ice (snow or
+! hail), the mixture. It also calculate the density of mixture.
+!
+!-----------------------------------------------------------------------
+!
+! AUTHOR:  Youngsun Jung, 5/30/2006
+!
+!-----------------------------------------------------------------------
+! Force explicit declarations.
+!-----------------------------------------------------------------------
+
+  IMPLICIT NONE
+
+!-----------------------------------------------------------------------
+! Declare variables.
+!-----------------------------------------------------------------------
+
+  REAL(r_kind) :: qr, qi, fo, density_ice
+  REAL(r_kind),INTENT(OUT) :: fracqr, fracqi, fm, fw, rhom
+
+  REAL(r_kind) :: fr
+
+!@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+!
+! Beginning of executable code...
+!
+!@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+  fr = 0.
+  fw = 0.
+  fracqr = 0.
+  fracqi = 0.
+  fm = 0.
+  rhom = 0.
+
+!-----------------------------------------------------------------------
+! Calculate the fraction of mleting ice (fr) based on the ratio between
+! qr and qi. fo is the maximum allowable fraction of melting snow.
+!-----------------------------------------------------------------------
+  IF (qr > 0. .AND. qi > 0.) THEN
+    fr = fo*(MIN(qi/qr,qr/qi))**.3
+  ENDIF
+
+!-----------------------------------------------------------------------
+! Calculate the faction of water and ice.
+! fracqr : the mass of water in the melting ice
+! fracqi : the mass of ice in the melting ice
+! fm     : total mass of melting ice
+! fw     : the fraction of water within melting ice
+! rhom   : density of mixture
+!-----------------------------------------------------------------------
+  fracqr = fr * qr
+  fracqi = fr * qi
+  fm = fracqr + fracqi
+
+  IF (fm .EQ. 0. .AND. qr > 0.) THEN
+    fw = 1.
+  ELSE IF (fm > 0.) THEN
+    fw = fracqr/fm
+  ENDIF
+
+  rhom = 1000.*fw**2. + (1.-fw**2.)*density_ice
+
+END SUBROUTINE fractionWater
+
+SUBROUTINE fractionWater_temperature_snow (qr,qi,density_ice,fm,fw,rhom,tair_C)
```

Comment:
The variable, qr, has been identified as unused.  Please remove unused variable.

Reply:
We changed the subroutine to remove qr as you suggested.

https://github.com/NOAA-EMC/GSI/pull/142/files#diff-fa0ecb3836c32fcbab1f64a2be08a2292b9c3414d702d475609b32f8205ffcf1R1786